File: 8-semester-pamo/exercise-3/Tipper/app/src/main/java/com/example/tipper/BmiXAxisFormatter.java
package com.example.tipper;

import com.github.mikephil.charting.components.AxisBase;
import com.github.mikephil.charting.formatter.ValueFormatter;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class BmiXAxisFormatter extends ValueFormatter {

    private SimpleDateFormat inputDateFormat = new SimpleDateFormat("yyyy-MM-dd");
    private SimpleDateFormat sdf = new SimpleDateFormat("dd.MM");
    private List<Date> dates;

    public BmiXAxisFormatter(String[] dateStrings) {
        this.dates = new ArrayList<>();
        for (String dateString : dateStrings) {
            try {
                Date date = inputDateFormat.parse(dateString);
                dates.add(date);
            } catch (ParseException e) {
                e.printStackTrace();
            }
        }
    }

    @Override
    public String getAxisLabel(float value, AxisBase axis) {
        int index = Math.round(value);
        if (index < 0 || index >= dates.size()) {
            return "";
        }
        return sdf.format(dates.get(index));
    }
}

File: 8-semester-pamo/exercise-4/Tipper/app/src/main/java/com/example/tipper/BmiXAxisFormatter.kt
package com.example.tipper

import com.github.mikephil.charting.components.AxisBase
import com.github.mikephil.charting.formatter.ValueFormatter
import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.*

class BmiXAxisFormatter(dateStrings: Array<String?>) : ValueFormatter() {
    private val inputDateFormat = SimpleDateFormat("yyyy-MM-dd")
    private val sdf = SimpleDateFormat("dd.MM")
    private val dates: MutableList<Date>

    init {
        dates = ArrayList()
        for (dateString in dateStrings) {
            try {
                val date = inputDateFormat.parse(dateString)
                dates.add(date)
            } catch (e: ParseException) {
                e.printStackTrace()
            }
        }
    }

    override fun getAxisLabel(value: Float, axis: AxisBase): String {
        val index = Math.round(value)
        return if (index < 0 || index >= dates.size) {
            ""
        } else sdf.format(dates[index])
    }
}